src/RpcServer/RpcServer.SmartContract.cs

```diff
@@ -141,79 +150,108 @@ private static JObject ToJson(System.Collections.Generic.IEnumerable<DataCache.T
         {
-            var ret = new Signers(_params.Select(u => new Signer()
+            var ret = _params.Select(u => new Signer
             {
                 Account = AddressToScriptHash(u["account"].AsString(), settings.AddressVersion),
                 Scopes = (WitnessScope)Enum.Parse(typeof(WitnessScope), u["scopes"]?.AsString()),
                 AllowedContracts = ((JArray)u["allowedcontracts"])?.Select(p => UInt160.Parse(p.AsString())).ToArray(),
                 AllowedGroups = ((JArray)u["allowedgroups"])?.Select(p => ECPoint.Parse(p.AsString(), ECCurve.Secp256r1)).ToArray(),
                 Rules = ((JArray)u["rules"])?.Select(WitnessRule.FromJson).ToArray(),
-            }).ToArray())
-            {
-                Witnesses = _params
-                    .Select(u => new
-                    {
-                        Invocation = u["invocation"]?.AsString(),
-                        Verification = u["verification"]?.AsString()
-                    })
-                    .Where(x => x.Invocation != null || x.Verification != null)
-                    .Select(x => new Witness()
-                    {
-                        InvocationScript = Convert.FromBase64String(x.Invocation ?? string.Empty),
-                        VerificationScript = Convert.FromBase64String(x.Verification ?? string.Empty)
-                    }).ToArray()
-            };
+            }).ToArray();
 
             // Validate format
 
-            _ = IO.Helper.ToByteArray(ret.GetSigners()).AsSerializableArray<Signer>();
+            _ = IO.Helper.ToByteArray(ret).AsSerializableArray<Signer>();
 
             return ret;
         }
 
+        private static Witness[] WitnessesFromJson(JArray _params)
+        {
+            return _params.Select(u => new
+            {
+                Invocation = u["invocation"]?.AsString(),
+                Verification = u["verification"]?.AsString()
+            }).Where(x => x.Invocation != null || x.Verification != null).Select(x => new Witness()
+            {
+                InvocationScript = Convert.FromBase64String(x.Invocation ?? string.Empty),
+                VerificationScript = Convert.FromBase64String(x.Verification ?? string.Empty)
+            }).ToArray();
+        }
+
         [RpcMethod]
         protected virtual JObject InvokeFunction(JArray _params)
         {
             UInt160 script_hash = UInt160.Parse(_params[0].AsString());
             string operation = _params[1].AsString();
             ContractParameter[] args = _params.Count >= 3 ? ((JArray)_params[2]).Select(p => ContractParameter.FromJson(p)).ToArray() : System.Array.Empty<ContractParameter>();
-            Signers signers = _params.Count >= 4 ? SignersFromJson((JArray)_params[3], system.Settings) : null;
+            Signer[] signers = _params.Count >= 4 ? SignersFromJson((JArray)_params[3], system.Settings) : null;
+            Witness[] witnesses = _params.Count >= 4 ? WitnessesFromJson((JArray)_params[3]) : null;
```

Comment:
Saved in the same _params as signers? (_params[3])

Reply:
Yes.